examples/common/builders.py

```diff
@@ -80,3 +83,50 @@ def build_dataloader(cfg, device_batch_size):
         return build_text_dataloader(cfg, device_batch_size)
     else:
         raise ValueError(f'Not sure how to build dataloader with config: {cfg}')
+
+def validate_cfg(eval_cfg):
+    assert "dataset_uri" in eval_cfg
+    assert "type" in eval_cfg
+    assert "num_fewshot" in eval_cfg
+    assert "batch_size" in eval_cfg
+    assert "metrics" in eval_cfg
+    assert "formatting_options" in eval_cfg
+    assert "prompt_string" in eval_cfg.get("formatting_options")
+    assert "example_delimiter" in eval_cfg.get("formatting_options")
+    assert "continuation_delimiter" in eval_cfg.get("formatting_options")
+    assert 'label' in eval_cfg
+
+def build_evaluators(cfg):
+    
+    tokenizer = TOKENIZER_REGISTRY[cfg.tokenizer.type](**cfg.tokenizer.args)
+    evaluators = []
+    logger_keys = []
+    for eval_cfg in cfg.icl_tasks:
+        validate_cfg(eval_cfg)
+        dataset_uri = eval_cfg.get("dataset_uri")
+        type = eval_cfg.get("type")
+        num_fewshots = eval_cfg.get("num_fewshot")
+        batch_size = eval_cfg.get("batch_size")
+        metrics = list(eval_cfg.get("metrics"))
+        prompt_string = eval_cfg.get("formatting_options").get("prompt_string")
+        example_delimiter = eval_cfg.get("formatting_options").get("example_delimiter")
+        continuation_delimiter = eval_cfg.get("formatting_options").get("continuation_delimiter")
```

Comment:
Code structure wise, could we turn this function `build_evaluators` into a function with args like 
```
build_evaluators(dataset_uri, type, num_fewshots, ...)
```
to separate it from the OmegaConf objects, which are just a user-defined format? That way, this function could be more general and would not require people to adopt the exact cfg structure we support.

(Also, I recognize that other functions in `builders.py` break this, but we have it on the roadmap to refactor those too :)